cohere_v3_vector/index-vectors-with-text-mapping.json

```diff
@@ -0,0 +1,34 @@
+{
+  "settings": {
+    {# non-serverless-index-settings-marker-start #}{%- if build_flavor != "serverless" or serverless_operator == true -%}
+      {% if preload_pagecache %}
+    "index.store.preload": [ "vec", "vex", "vem"],
+      {% endif %}
+    "index.number_of_shards": {{number_of_shards | default(1)}},
+    "index.number_of_replicas": {{number_of_replicas | default(0)}}
+    {%- endif -%}{# non-serverless-index-settings-marker-end #}
+  },
+  "mappings": {
+    "properties": {
+      "docid": {
+        "type": "keyword"
+      },
+      "title": {
+        "type": "text"
+      },
+      "text": {
+        "type": "text"
+      },
+      "emb": {
+        "type": "dense_vector",
+        "element_type": "float",
+        "dims": 1024,
+        "index": true,
+        "similarity": "max_inner_product",
```

Comment:
The embeddings are normalized already so we can use `dot_product` directly

Reply:
Annoyingly it looks like they aren't all normalized, ES is complaining that a bunch of vectors aren't normalized, and while I can confirm the queries are definitely unit-length vectors as the embeds came from the Cohere API which serves normalized vectors only, it looks like the dataset isn't.